src/actions/transferActions.js

```diff
@@ -721,14 +721,29 @@ function submitTx (txRequest: {
   txFeePerByte: ?number,
 }) {
   return (dispatch: Function, getState: Function) => {
+    const { walletType, fromWallet, cryptoType } = txRequest
     let accountInfo = {}
-    if (txRequest.cryptoType === 'bitcoin') {
-      accountInfo = getState().walletReducer.wallet[txRequest.walletType].crypto[txRequest.cryptoType][0]
+    if (cryptoType === 'bitcoin') {
+      accountInfo = getState().walletReducer.wallet[walletType].crypto[cryptoType][0]
     }
     return dispatch({
       type: 'SUBMIT_TX',
       payload: _submitTx(txRequest, accountInfo)
-    }).then(() => dispatch(goToStep('send', 1)))
+    }).then(() => {
+      dispatch(goToStep('send', 1))
+      const googleId = getState().userReducer.profile.googleId
+      const address = cryptoType === 'bitcoin' && walletType === 'ledger'
+        ? accountInfo.xpub
+        : fromWallet.crypto[cryptoType][0].address
+      dispatch(setLastUsedAddress({ googleId, walletType, address }))
+    })
```

Comment:
two dispatches should be connected with 
```
.then(() =>dispatch())
```

The order should be setLastUsedAddress => goToStep, this ensures setLastUsedAddress is always invoked.
Move googleId and address to the top, also check if use has logged in, otherwise googleId === null